### PR TITLE
Replace Cloud SDK Pipeline with General Purpose Pipeline

### PR DESCRIPTION
**Pull request by fwilhe**
## Proposed Changes

Update the Pipeline template to project "Piper" General Purpose Pipeline.

See https://github.com/SAP/cloud-s4-sdk-pipeline/blob/master/gpp-guide.md for more background.

## Checklist

- [ ] I have added or adjusted tests that prove my fix is effective or that my feature works
- [ ] I have added or adjusted documentation

## Further Comments

If the changes are complex, please discuss what alternatives you have evaluated and why you picked this particular solution. Please make sure to highlight any trade-offs that we should know about.

### CODE REVIEW

File: src/templates/init/.pipeline/config.yml.mu [hunk 1]
```diff
@@ -1,39 +1,54 @@
...
+  buildExecute:
+    npmRunScripts:
+      - ci-build
+      - ci-package
 
 ### Stage-specific configuration
 stages:
-#  productionDeployment:
+  Integration:
+    install: true
```
Comment (daniel-kurzynski) [inline src/templates/init/.pipeline/config.yml.mu:40]: this should not be required

**Review by daniel-kurzynski — APPROVED** [review]

**Review by marikaner — COMMENTED** [review]
LGTM. @jjtang1985 can you please check, whey there are checks that have not run?

File: src/templates/init/.pipeline/config.yml.mu [hunk 2]
```diff
@@ -1,39 +1,53 @@
...
+  artifactPrepareVersion:
+    versioningType: cloud_noTag
+
+  pipelineStashFilesAfterBuild:
+    stashIncludes:
+      deploymentFolder: 'deployment/**'
+
+  cloudFoundryDeploy:
+    stashContent:
+      - 'deployDescriptor'
```
Comment (KevinHudemann) [inline src/templates/init/.pipeline/config.yml.mu:28]: Minor: This one was already added to the metadata: https://github.com/SAP/jenkins-library/blob/master/resources/metadata/cloudFoundryDeploy.yaml#L18

**Review by KevinHudemann — APPROVED** [review]
👍  only one minor comment

File: src/templates/init/.pipeline/config.yml.mu (hunk 2 above)
Comment (daniel-kurzynski) [reply src/templates/init/.pipeline/config.yml.mu:28]: But I think with adding a key stashContent it will ignore it, doesn't it?

Comment (fwilhe) [reply src/templates/init/.pipeline/config.yml.mu:28]: That was my assumption too

Comment (fwilhe) [reply src/templates/init/.pipeline/config.yml.mu:28]: image

I've tried it out, and it looks like it is not unstashed if I remove it here.

**Comment by jjtang1985** [comment]
As discussed the additional checks related to the node 14 come from this PR:https://github.com/SAP/cloud-sdk-cli/pull/240

**Comment by fwilhe** [comment]
> As discussed the additional checks related to the node 14 come from this PR:#240

Ok, is green now. Thanks for your support 👍

**Pull request merged**